daisy/snapshot_test.go

```diff
@@ -0,0 +1,86 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package daisy
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"google.golang.org/api/compute/v1"
+)
+
+func TestSnapshotPopulate(t *testing.T) {
+	w := testWorkflow()
+
+	tests := []struct {
+		desc      string
+		ss        *Snapshot
+		shouldErr bool
+	}{
+		{"default case", &Snapshot{}, false},
+	}
+
+	for testNum, tt := range tests {
+		s, _ := w.NewStep("s" + strconv.Itoa(testNum))
+		err := tt.ss.populate(context.Background(), s)
```

Comment:
Minor, and feel free to ignore for this PR, but a neat feature that I found recently (you may have already seen it) is `t.Run`, which creates a named subtest. In the test output, you'll see a nice row for each case in your table.

- https://github.com/golang/go/wiki/TableDrivenTests
- https://golang.org/pkg/testing/#T.Run